xgcm/test/test_grid.py

```diff
@@ -658,3 +659,41 @@ def test_multi_axis_input(all_datasets, func, periodic, boundary):
             serial = getattr(grid, func)(serial, axis, boundary=boundary_axis)
         full = getattr(grid, func)(ds[varname], axes, boundary=boundary)
         xr.testing.assert_allclose(serial, full)
+
+
+@pytest.mark.parametrize(
+    "funcname",
+    [
+        "diff",
+        "interp",
+        "min",
+        "max",
+        "integrate",
+        "average",
+        "cumsum",
+        "cumint",
+        "derivative",
+    ],
+)
```

Comment:
It is probably not strictly necessary but could you parameterize the grid type as well?
```python
@pytest.mark.parametrize("gridtype", ["B", "C"])
def test_keep_coords(funcname, gridtype):
    ds, coords, metrics = datasets_grid_metric(gridtype)
```
That way we make sure that all cases are covered.

Reply:
Awesome thanks.

Reply:
done